base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -1888,97 +1892,159 @@ where
         Ok(())
     }
 
-    /// Attempt to scan and then rewind all of the given transaction outputs into unblinded outputs based on known
-    /// pubkeys
+    #[allow(clippy::too_many_lines)]
     fn scan_outputs_for_one_sided_payments(
         &mut self,
         outputs: Vec<TransactionOutput>,
     ) -> Result<Vec<RecoveredOutput>, OutputManagerError> {
-        let known_one_sided_payment_scripts: Vec<KnownOneSidedPaymentScript> =
-            self.resources.db.get_all_known_one_sided_payment_scripts()?;
-
-        let mut rewound_outputs: Vec<RecoveredOutput> = Vec::new();
-        for output in outputs {
-            let position = known_one_sided_payment_scripts
-                .iter()
-                .position(|known_one_sided_script| known_one_sided_script.script == output.script);
-            if let Some(i) = position {
-                let spending_key = PrivateKey::from_bytes(
-                    CommsPublicKey::shared_secret(
-                        &known_one_sided_payment_scripts[i].private_key,
-                        &output.sender_offset_public_key,
-                    )
-                    .as_bytes(),
-                )?;
-                let rewind_blinding_key = PrivateKey::from_bytes(&hash_secret_key(&spending_key))?;
-                let encryption_key = PrivateKey::from_bytes(&hash_secret_key(&rewind_blinding_key))?;
-                let committed_value =
-                    EncryptedValue::decrypt_value(&encryption_key, &output.commitment, &output.encrypted_value);
-                if let Ok(committed_value) = committed_value {
-                    let blinding_factor =
-                        output.recover_mask(&self.resources.factories.range_proof, &rewind_blinding_key)?;
-                    if output.verify_mask(
-                        &self.resources.factories.range_proof,
-                        &blinding_factor,
-                        committed_value.into(),
-                    )? {
-                        let rewound_output = UnblindedOutput::new(
-                            output.version,
-                            committed_value,
-                            blinding_factor.clone(),
-                            output.features,
-                            known_one_sided_payment_scripts[i].script.clone(),
-                            known_one_sided_payment_scripts[i].input.clone(),
-                            known_one_sided_payment_scripts[i].private_key.clone(),
-                            output.sender_offset_public_key,
-                            output.metadata_signature,
-                            known_one_sided_payment_scripts[i].script_lock_height,
-                            output.covenant,
-                            output.encrypted_value,
-                        );
+        // ----------------------------------------------------------------------------
+        // scanning
```

Comment:
This function is now super long, 
Rather than make these comments, 
I would split the function into two.
One for scanning(this one can also potentially be divided into two) and one for processing.

Reply:
I thought about it, though the last function that adds scanned outputs would not be used for anything else. I have a TODO in place to make it shorter instead.

Reply:
Actually, you're right. I'll split it into two, because the 2 can be (possibly) useful for independent use.